src/cli/run.js

```diff
@@ -0,0 +1,128 @@
+// @flow
+/* eslint-disable no-console */
+import type {IntrospectionQuery} from 'graphql/utilities/introspectionQuery';
+
+import {generateTypeFiles, processPragmas} from '../generateTypeFiles';
+import {processFiles} from '../parser/parse';
+import {resolveDocuments} from '../parser/resolve';
+import {schemaFromIntrospectionData} from '../schemaFromIntrospectionData';
+import {loadConfigFile} from './config';
+
+import {addTypenameToDocument} from 'apollo-utilities'; // eslint-disable-line flowtype-errors/uncovered
+
+import {execSync} from 'child_process';
+import fs, {readFileSync} from 'fs';
+import {buildClientSchema, type DocumentNode} from 'graphql';
+import {print} from 'graphql/language/printer';
+import {validate} from 'graphql/validation';
+import path from 'path';
+
+const findGraphqlTagReferences = (root: string): Array<string> => {
+    const response = execSync(
+        "git grep -I --word-regexp --name-only --fixed-strings 'graphql-tag' -- '*.js' '*.jsx'",
+        {
+            encoding: 'utf8',
+            cwd: root,
+        },
+    );
+    return response
+        .trim()
+        .split('\n')
+        .map((relative) => path.join(root, relative));
+};
+
+const [_, __, configFile, ...cliFiles] = process.argv;
+
+const config = loadConfigFile(configFile);
+
+const inputFiles = cliFiles.length
+    ? cliFiles
+    : findGraphqlTagReferences(process.cwd());
+
+const files = processFiles(inputFiles, (f) => readFileSync(f, 'utf8'));
+
+let filesHadErrors = false;
+Object.keys(files).forEach((key) => {
+    const file = files[key];
+    if (file.errors.length) {
+        filesHadErrors = true;
+        console.log(`Errors in ${file.path}`);
+        file.errors.forEach((error) => {
+            console.log(` - ${error.message}`);
```

Comment:
I think all these `console.log` messages should go to `console.error` so they're less likely to interfere with pipelined commands (in case something that invokes this code expects its stdout to be machine-readable)